dashboard/apps/prototype/views.py

```diff
@@ -1,155 +1,55 @@
-        for key, val in props.items():
-            months.append(month_str)
-            keys.append(key)
-            values.append(val)
-    data = {
-        'months': months,
-        'keys': keys,
-        'values': values
-    }
-    label = CatAttr(columns='months', sort=False)
-    bar = Bar(data, values='values', label=label, stack='keys',
-              color=['dodgerblue', 'lightblue'],
-              title="Forecast FTE Civil Servant/Contractor split",
-              legend='top_right', **STYLE)
-    # TODO this does not feel right, but bokeh doesn't present
-    # a better way to do it.
-    bar._yaxis.formatter = NumeralTickFormatter(format='0.00')
-    return bar
-
-
-def project_timeline(timeline):
-    padding = timedelta(days=60)
-    start_date = timeline[list(PROJECT_PHASES.keys())[0]]['date']
-    end_date = max(timeline[list(PROJECT_PHASES.keys())[-1]]['date'],
-                   datetime.now())
-
-    x = [datetime(year=period.year, month=period.month, day=1) for period in
-         pd.period_range((start_date - padding).strftime('%Y-%m-%d'),
-                         (end_date + padding).strftime('%Y-%m-%d'),
-                         freq='M')
-         ]
-    p = figure(title='Timeline', plot_width=400,
-               plot_height=120, responsive=True,
-               x_axis_type="datetime", y_range=(-2, 8), logo=None)
-    p.line(x=x, y=[0] * len(x))
-    p.yaxis.visible = None
-    p.ygrid.grid_line_color = None
-    p.outline_line_color = None
-    p.xaxis.formatter = DatetimeTickFormatter(formats=dict(
-        days=["%b %y"],
-        months=["%b %y"],
-        years=["%b %y"],
-    ))
-
-    for phase, props in PROJECT_PHASES.items():
-        date = timeline[phase]['date']
-        p.diamond([date], [0], size=10, color=props['color'],
-                  alpha=0.5, legend=phase)
-
-    p.circle([datetime.now()], [0], size=10,
-             color="black", alpha=0.5, legend="today")
-    p.legend.orientation = 'top_left'
-    p.legend.background_fill_alpha = 0.5
-    return p
-
-
-@login_required
-def index(request):
-    script, div = components({
-        'MonthlySpend': monthly_spendings(
-            data_generator.gen_monthly_spendings()),
-        'CumulativeSpend': line(),
-        'ForecastFTE': forecast_stack(data_generator.gen_forcast()),
-        'Timeline': project_timeline(data_generator.gen_project_timeline()),
-    })
-    today = datetime.now().strftime('%d %b %Y')
-    context = dict(
-        js_resource=INLINE.render_js(),
-        css_resource=INLINE.render_css(),
-        script=script,
-        div=div,
-        today=today,
-    )
-    return render(request, 'index.html', context)
+import json
+import random
+from django.views.generic import View, TemplateView
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponse, JsonResponse, HttpResponseRedirect
+
+
+class Index(TemplateView):
+
+    template_name = 'index.html'
+
+
+class DataResponse(View):
```

Comment:
it's just a style thing. in idiomatic python, you don't need to use class for these views. 

to me this is a good guide on python as an OO language http://docs.python-guide.org/en/latest/writing/structure/#object-oriented-programming

this is a nice talk from a well know python hacker about the same issue. he has gone a little far 😄  https://www.youtube.com/watch?v=o9pEzgHorH0